src/lib/yacc/grammar.rs

```diff
@@ -1035,4 +1136,42 @@ mod test {
         find("C", vec![vec!["x"]]);
         find("D", vec![vec!["y", "x"], vec!["y", "z"]]);
     }
+
+    #[test]
+    fn test_nonterm_max_costs1() {
+        let grm = yacc_grm(YaccKind::Original, &"
+            %start A
+            %%
+            A: A B | ;
+            B: C | D | E;
+            C: 'x' B | 'x';
+            D: 'y' B | 'y' 'z';
+            E: 'x' A | 'x' 'y';
+          ".to_string()).unwrap();
+
+        let scores = nonterm_max_costs(&grm, &vec![1, 1, 1]);
+        assert_eq!(scores[usize::from(grm.nonterm_idx(&"A").unwrap())], u64::max_value());
+        assert_eq!(scores[usize::from(grm.nonterm_idx(&"B").unwrap())], u64::max_value());
```

Comment:
Couldn't `NTIdx` define a method which is akin to `u64::max_value`?

E.g.:
```
assert_eq!(scores[grm.nonterm_idx("C").unwrap()], NTIdx::infinitiy());
```

Reply:
We definitely don't want `NTIdx` to expose that. The use of `u64::max_value()` is a purely internal "trick" to cfgrammar, and I don't want users to think about that!